### PR TITLE
adding placeholders for other plugin

### PR DESCRIPTION
**Pull request by alan67160**
`%servertutorialplus_getTutorial%`
get the current tutorial ID
return null if no tutorial are playing or player offline
`%servertutorialplus_getPoints%`
get the current point index
return null if no tutorial are playing or player offline
`%servertutorialplus_getMaxPoints%`
get the max point index for the current tutorial
return null if no tutorial are playing or player offline

### CODE REVIEW

**Comment by alan67160** [comment]
btw you can request to add this plugin on https://github.com/PlaceholderAPI/PlaceholderAPI/wiki/Plugins-using-PlaceholderAPI#s

File: src/main/java/nl/martenm/servertutorialplus/managers/PlaceholderapiManager.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import java.util.HashMap;
+import java.util.UUID;
+
+import org.bukkit.OfflinePlayer;
+
+import me.clip.placeholderapi.expansion.PlaceholderExpansion;
+import nl.martenm.servertutorialplus.ServerTutorialPlus;
+import nl.martenm.servertutorialplus.objects.TutorialController;
+
+public class PlaceholderapiManager extends PlaceholderExpansion {
```
Comment (MartenM) [inline src/main/java/nl/martenm/servertutorialplus/managers/PlaceholderapiManager.java:12]: Please change the name to something like: ServertutorialPlaceholderExpansion.
It's not really an API manager class.

**Review by MartenM — CHANGES_REQUESTED** [review]
Solid contribution apart from that one issue!

File: src/main/java/nl/martenm/servertutorialplus/managers/PlaceholderapiManager.java (hunk 1 above)
Comment (alan67160) [reply src/main/java/nl/martenm/servertutorialplus/managers/PlaceholderapiManager.java:12]: actually, my original idea name it as "PlaceholderapiHandler.java", but I don't know where should I put it on
however, you are the original author of this plugin, I will respect your opinion

**Comment by alan67160** [comment]
@MartenM renamed, please check again

**Comment by alan67160** [comment]
@MartenM If I want to add a placeholder for check if the player has played the tutorial or not, do you think will it wold be better if a linkedhashmap added for caching? or just rely on hikari cache?

**Comment by alan67160** [comment]
idea is to on PlayerJoinEvent and TutorialEndEvent get data from mysql and save to linkedhashmap
on placeholder request get data from mysql

**Comment by MartenM** [comment]
A hashmap would be preferable since placeholders should be resolved pretty quickly.
However, I would suggest waiting with this feature as I want to merge this simple one first!

**Comment by alan67160** [comment]
OK, then that will be my next new PR

**Comment by MartenM** [comment]
I changed some things such that it makes more sense for the end users. Also moved the hooks to the API.
This has now been released https://www.spigotmc.org/resources/server-tutorial-plus.37741/update?update=518708

**Comment by alan67160** [comment]
ok, but just let you know If you return null to papi request, it will just replace to empty string automatically

**Comment by MartenM** [comment]
When returning null I got just the placeholder but no replacement which might confuse users.
It won't replace it, since it doesn't know it should or should not replace it.

**Pull request merged**